core/blockchain.go

```diff
@@ -1344,6 +1344,21 @@ func (bc *BlockChain) WriteBlockWithState(block *types.Block, receipts []*types.
 		// Split same-difficulty blocks by number
 		reorg = block.NumberU64() > currentBlock.NumberU64()
 	}
+
+	// if reorg {
```

Comment:
This is the ETH fix. We shall ultimately have this workflow, but due to code has diverged significantly between ETH and XDC, it's best to have it in a different PR with more investigations